ImageConverter/ImageConverter/Ppm/PpmImageReader.cs

```diff
@@ -0,0 +1,62 @@
+﻿using System.Text;
+
+namespace ImageConverter.Ppm;
+
+public class PpmImageReader : IImageReader
+{
+    public Image Read(string source)
+    {
+        Pixel[,] pixelmap;
+        using (StreamReader streamReader = new StreamReader(source))
+        {
+            string fileFormatNumber = ReadUntilDelimiter(streamReader);
+            if (fileFormatNumber != "P3")
+                throw new ArgumentException("This file is not plain PPM format");
+            int width = int.Parse(ReadUntilDelimiter(streamReader));
+            int height = int.Parse(ReadUntilDelimiter(streamReader));
+            int colorDepth = int.Parse(ReadUntilDelimiter(streamReader));
+            pixelmap = new Pixel[height, width];
+            for (int i = 0; i < height; i++)
+            {
+                for (int j = 0; j < width; j++)
+                {
+                    string red = ReadUntilDelimiter(streamReader);
+                    byte redValue = byte.Parse(red);
+                    string green = ReadUntilDelimiter(streamReader);
+                    byte greenValue = byte.Parse(green);
+                    string blue = ReadUntilDelimiter(streamReader);
+                    byte blueValue = byte.Parse(blue);
+                    pixelmap[i, j] = new Pixel(redValue, greenValue, blueValue);
+                }
+            }
+        }
+        return new Image(pixelmap);
+    }
+
+    private string ReadUntilDelimiter(StreamReader stream)
+    {
+        StringBuilder result = new();
+        char currentChar = (char)stream.Read();
+        while (!char.IsLetterOrDigit(currentChar))
```

Comment:
Check to be exactly digit so that letters are ignored or throw an exception that file is corrupted

Reply:
no need because after in the read method, I try to parse it, which throws an exception if there will be letter, there must be a number